app/src/main/java/org/mozilla/tv/firefox/toolbar/ToolbarViewModel.kt

```diff
@@ -70,15 +72,27 @@ open class ToolbarViewModel(
         disableDesktopModeWhenHostChanges()
     }
 
-    fun backButtonClicked() = sessionRepo.exitFullScreenIfPossibleAndBack()
+    @UiThread
+    fun backButtonClicked() {
+        sessionRepo.exitFullScreenIfPossibleAndBack()
+        closeOverlay()
+    }
 
-    fun forwardButtonClicked() = sessionRepo.goForward()
+    @UiThread
+    fun forwardButtonClicked() {
+        sessionRepo.goForward()
+        closeOverlay()
```

Comment:
hm, if every action closes the overlay, I'm wondering if it's necessary to have it in each one? But maybe this is just easier and it's not worth adding another code path to generalize that "close on click" behavior, so I'm fine with it. Just concerned that we might forget to add this with new functionality.

Reply:
Overlay closing should also be moved to the overlay VM when possible, so this is fairly temporary.  As it is, it should be easy to remove.